OpenRA.Mods.Common/Traits/Player/BulkProductionQueue.cs

```diff
@@ -0,0 +1,367 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2024 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Attach this to the player actor (not a building!) to define a new shared build queue.",
+	"Allow you to build multiple actors before delivery",
+	"Will work together with the ProductionProduction: trait on the actor that actually does the production.",
+	"You will also want to add PrimaryBuildings: to let the user choose where new units should exit.")]
+	public class BulkProductionQueueInfo : ProductionQueueInfo, Requires<TechTreeInfo>, Requires<PlayerResourcesInfo>
+	{
+		[Desc("Maximum order capacity")]
+		public readonly int MaxCapacity = 6;
+
+		[Desc("Delivery delay in ticks")]
+		public readonly int DeliveryDelay = 1500;
+
+		[NotificationReference("Speech")]
+		[Desc("Notification played when deliver started")]
+		public readonly string StartDeliveryAudio = null;
+
+		[Desc("Return funds if delivery fails")]
+		public readonly bool RefundUndeliveredActors = true;
+
+		[TranslationReference(optional: true)]
+		[Desc("Notification displayed when deliver started")]
+		public readonly string StartDeliveryNotification = null;
+
+		[NotificationReference("Speech")]
+		[Desc("Notifications played while frigate is on the way. Starting with first. Ending with last one when delivery actor is spawned on the map")]
+		public readonly string[] DeliveryProgressNotifications = Array.Empty<string>();
+
+		public override object Create(ActorInitializer init) { return new BulkProductionQueue(init, this); }
+	}
+
+	public class BulkProductionQueue : ProductionQueue
+	{
+		static readonly ActorInfo[] NoItems = Array.Empty<ActorInfo>();
+
+		readonly Actor self;
+		readonly BulkProductionQueueInfo info;
+
+		// Used in refunds. Key is resources, Value is cash
+		protected readonly List<(ActorInfo Actor, int Resources, int Cash)> ActorsCost = new();
+		protected readonly List<ActorInfo> ActorsReadyForDelivery = new();
+		int deliveryDelay = 0;
```

Comment:
```suggestion
		public int DeliveryDelay { get; private set; } = 0;
```

and remove the GetDeliveryTime() method